lib/elastic_searcher.rb

```diff
@@ -59,6 +58,9 @@ def search_definition # rubocop:disable Metrics/MethodLength
         end
       end
 
+      # only return gems that are not yanked unless filter is used
+      post_filter { term yanked: false } unless query_str.gsub(/[[:space:]]/, "").include? "yanked:true"
```

Comment:
Found an issue with this. Using existing query with yanked filter removed means, gems where search term exists in summary and description are also shown. We want to show yanked gem only if query matches name only.

Reply:
Ah, correct. Missed it.

Reply:
for name too, we only want to show if there is exact match I think this will need separate query.